### PR TITLE
New Cut in Chisel

### PR DESCRIPTION
**Pull request by IveanEx**
This PR aims to work with @xiaoling-yi to investigate and fix the long critical path problem exists in our circuit, so that we can goes to high frequency. 

To fulfill this target, this PR implements a new **DataCut** component, which datapath is implemented by shift registers and Controller controls the behavior of the system to compliant with Valid / Ready signal handshaking protocol. 

To ease the usage, the new operator is introduced to insert it: **-\\>**, **-\\\\>** and **-\\\\\\>**. 

Please be aware that the Control Signal is not cut. To also cut the Control signal, you should use **-|>** or **-||>** combined with **-\\>**, **-\\\\>** and **-\\\\\\>**. The former one (**-|>** or **-||>**) is not cannot be retimed, because it is implemented by Ping-Pong architecture.

### CODE REVIEW

**Review by xiaoling-yi — APPROVED** [review]

**Comment by rgantonio** [comment]
PR description, please! 😄

**Comment by IveanEx** [comment]
> PR description, please! 😄

Available now!

**Comment by rgantonio** [comment]
That's a neat trick! Okay go go go!

File: hw/chisel/src/main/scala/snax/utils/CustomOperators.scala [hunk 1]
```diff
@@ -9,6 +9,56 @@ import chisel3.reflect.DataMirror
...
+//   })
+//   val cuts = Seq.fill(delay)(Module(new Queue(gen, 1, pipe = true)))
+
+//   io.in <> cuts.head.io.enq
+//   cuts.zip(cuts.tail).foreach { case (left, right) =>
+//     left.io.deq <> right.io.enq
+//   }
+//   cuts.last.io.deq <> io.out
+
+// }
```
Comment (rgantonio) [inline hw/chisel/src/main/scala/snax/utils/CustomOperators.scala:27]: Do you need these comments?

**Review by rgantonio — APPROVED** [review]
Go go go!

File: hw/chisel/src/main/scala/snax/utils/CustomOperators.scala (hunk 1 above)
Comment (IveanEx) [reply hw/chisel/src/main/scala/snax/utils/CustomOperators.scala]: Removed

**Pull request merged**